pkg/cstor/volume/targetserver/server.go

```diff
@@ -0,0 +1,184 @@
+			if strings.HasSuffix(string(completeBytes), endOfLine) {
+				// Since we are ending only when \r\n bytes are present in
+				// completeBytes to extract \r\n need to perform below steps
+				lines := strings.Split(string(completeBytes), endOfLine)
+				// here we are excluding \r\n
+				for _, line := range lines {
+					if len(line) != 0 {
+						req = append(req, line)
+					}
+				}
+				req = append(req, string(completeBytes))
+				break
+			}
+		}
+	}
+	return fmt.Sprintf("%s", req), nil
+}
+
+//TODO: Once tested need to remove commented code
+
+// GetRequiredData returns error if doesn't have json format
+func GetJsonData(data string) (string, error) {
+	jsonBeginIndex := strings.Index(data, "{")
+	jsonEndIndex := strings.LastIndex(data, "}")
+	if jsonBeginIndex >= jsonEndIndex {
+		return "", errors.Errorf("failed to parse the data got: %s", data)
+	}
+	return data[jsonBeginIndex : jsonEndIndex+1], nil
+}
+
+//ServeRequest process the request from the client
+func ServeRequest(conn net.Conn, kubeClient *cstorv1alpha1.Kubeclient) {
+	var err error
+	var readData, filteredData string
+	defer func() {
+		if err != nil {
+			_, er := conn.Write([]byte(respErr + endOfLine))
+			if er != nil {
+				klog.Errorf("failed to inform to client")
+			}
+		} else {
+			_, er := conn.Write([]byte(respOk + endOfLine))
+			if er != nil {
+				klog.Errorf("failed to inform to client")
+			}
+		}
+		conn.Close()
+	}()
+	readData, err = Reader(conn)
+	if err != nil {
+		klog.Errorf("failed to read data: {%v}", err)
+		return
+	}
+	//TODO: Not required but need to test below function
+	filteredData, err = GetJsonData(readData)
+	if err != nil {
+		klog.Errorf("failed to get required information: {%v}", err)
+		return
+	}
+	replicationData := &cstorv1alpha1.CStorVolumeReplication{}
+	err = json.Unmarshal([]byte(filteredData), replicationData)
+	if err != nil {
+		klog.Errorf("failed to unmarshal replication data {%v}", err)
+		return
+	}
+	csc := &cstorv1alpha1.CStorVolumeConfig{
+		CStorVolumeReplication: replicationData,
+		Kubeclient:             kubeClient,
+	}
+	err = csc.UpdateCVWithReplicationDetails()
+	if err != nil {
+		klog.Errorf("failed to update cstorvolume {%s} with details {%v}"+
+			" error: {%v}", csc.VolumeName,
+			replicationData, err)
+		return
+	}
+}
+
+// StartTargetServer starts the UnixDomainServer
+func StartTargetServer(kubeConfig string) {
+
+	klog.Info("Starting unix domain server")
+	if err := os.RemoveAll(string(volumeMgmtUnixSock)); err != nil {
+		klog.Fatalf("failed to clear path: {%v}", err)
+	}
+
+	listen, err := net.Listen("unix", volumeMgmtUnixSock)
+	if err != nil {
+		klog.Fatalf("listen error: {%v}", err)
+	}
+
+	//TODO: Remove hard coded ENV
+	namespace := env.Get("CSTOR_TARGET_NAMESPACE")
+	if namespace == "" {
+		klog.Fatalf("failed to get volume namespace empty value for env %s",
+			"CStorVolumeReplication",
+		)
+	}
+
+	sigc := make(chan os.Signal, 1)
+	signal.Notify(sigc, os.Interrupt, syscall.SIGTERM)
```

Comment:
why for Interrupt?

Reply:
should we consider SIGKILL?

Reply:
or you better remove all the code related to signals

Reply:
If remove the code related to signals then we are not closing the listener connection which will be the problem for the next listener(i.e when the program starts again) it might throw an error address already in use.

Reply:
Removed signal code `os.RemoveAll` kept as is